File: 250515/아름다운 수열 2/beautiful-sequence-2.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int M = sc.nextInt();
        int[] A = new int[N];
        for (int i = 0; i < N; i++)
            A[i] = sc.nextInt();
        // int[] B = new int[M];
        Set<Integer> set = new HashSet<>();
        for (int i = 0; i < M; i++)
            // B[i] = sc.nextInt();
            set.add(sc.nextInt());
        // Please write your code here.
        
        int count = 0;
        for (int i = 0; i <= N-M; i++) {
            Set<Integer> setCopy = new HashSet<>(set);
            for (int j = i; j < i+M; j++) {
                if (setCopy.contains(A[j])) {
                    setCopy.remove(A[j]);
                } else {
                    break;
                }
                if (j == i+M-1) count++;
            }
        }

        System.out.print(count);
    }
}

// 

File: 251024/아름다운 수열 2/beautiful-sequence-2.py
N, M = map(int, input().split())
A = list(map(int, input().split()))
B = list(map(int, input().split()))
B.sort()

# Please write your code here.

cnt = 0
if M>N :
    print(0)
else :
    for i in range(N-(M-1)) :
        s = []

        for j in range(i,i+M) :
            s.append(A[j])
        s.sort()
        if s == B :
            cnt += 1
    print(cnt)
    
